src/mlpack/core/data/load_image_impl.hpp

```diff
@@ -0,0 +1,202 @@
+/**
+ * @file load_image_impl.hpp
+ * @author Mehul Kumar Nirala
+ *
+ * An image loading utillity
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_LOAD_IMAGE_IMPL_HPP
+#define MLPACK_CORE_DATA_LOAD_IMAGE_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "load_image.hpp"
+
+namespace mlpack {
+namespace data {
+
+LoadImage::LoadImage():
+        matrixWidth(-1),
+        matrixHeight(-1),
+        channels(3)
+{
+  fileTypes.clear();
+
+  // Declare supported file types.
+  fileTypes.insert(fileTypes.end(),
+    {"jpg", "png", "tga", "bmp", "psd", "gif", "hdr", "pic", "pnm"});
+}
+
+LoadImage::LoadImage(int width,
+          int height,
+          int channels):
+          matrixWidth(width),
+          matrixHeight(height),
+          channels(channels)
+{
+  fileTypes.clear();
+
+  // Declare supported file types.
+  fileTypes.insert(fileTypes.end(),
+    {"jpg", "png", "tga", "bmp", "psd", "gif", "hdr", "pic", "pnm"});
+}
+
+LoadImage::~LoadImage()
+{
+  // Do nothing.
+}
+
+bool LoadImage::IsImageFile(std::string& fileName)
+{
+  // Iterate over all supported file types.
+  for (auto extension : fileTypes){
+    std::transform(extension.begin(), extension.end(), extension.begin(),
+                    ::tolower);
+    if (extension == Extension(fileName))
+      return true;
+  }
+  return false;
+}
+
+bool LoadImage::Load(std::string& fileName,
+                      arma::Mat<unsigned char>&& outputMatrix,
+                      int *width,
+                      int *height,
+                      int *channels)
+{
+  unsigned char *image;
+
+  if (!IsImageFile(fileName))
+  {
+    std::ostringstream oss;
+    oss << "File type " << Extension(fileName) << " not supported.\n";
+    oss << "Currently it supports ";
+    for (auto extension : fileTypes)
+      oss << " " << extension;
+    oss << std::endl;
+    throw std::runtime_error(oss.str());
+  }
+
+  stbi_set_flip_vertically_on_load(true);
```

Comment:
Should this be an option, a user could set?

Reply:
Okay